free-tunes/Controllers/LoginViewController.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  LoginViewController.swift
+//  free-tunes
+//
+//  Created by Tshwarelo Mafaralala on 2022/02/28.
+//
+
+import UIKit
+
+class LoginViewController: UIViewController {
+
+    @IBOutlet weak var headerText: UILabel!
+    @IBOutlet weak var usernameInput: UITextField!
+    @IBOutlet weak var passwordInput: UITextField!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupInputFields()
+        headerText.buildCustomLabel(labelFirstPart: "free", labelSecondPart: "Tunes")
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        self.navigationController?.setNavigationBarHidden(false, animated: animated)
+        super.viewWillAppear(animated)
+    }
+
+    func setupInputFields() {
+        usernameInput.applyDefaultStyle(withName: "Username")
+        usernameInput.delegate = self
+        usernameInput.delegate = self
+        passwordInput.applyDefaultStyle(withName: "Password")
+        passwordInput.delegate = self
+        passwordInput.delegate = self
+    }
+
+    func loginRequest() -> Bool {
+        if usernameInput.text != "Admin" || passwordInput.text != "TestPass123" {
+            let alert = UIAlertController()
+            alert.displayErrorAlert(alertTitle: "Invalid credentials.",
+                                    alertMessage: "Incorrect Username or Password.",
+                                    alertActionTitle: "Try again" ,
+                                    alertDelegate: self)
+            return false
+        }
+        return true
+    }
+
+    override func shouldPerformSegue(withIdentifier identifier: String, sender: Any?) -> Bool {
+
+        if usernameInput.text != "" && passwordInput.text != "" {
+            return loginRequest()
+        }
+        usernameInput.emptyFieldError()
+        passwordInput.emptyFieldError()
+        return false
+    }
+
+}
+
+extension LoginViewController: UITextFieldDelegate {
+
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        guard let name = textField.layer.name else {
+            return
+        }
+        if name == "Username" || name == "Password" {
+            textField.layer.borderColor = UIColor(named: "AppPrimaryColor")?.cgColor
```

Comment:
I would suggest wrapping this in an extension so that you do not constantly have to write this code every time you want rto change the textfields color

Reply:
I'm not sure I fully understand which code would need to be re written, do you mind elaborating further.

Reply:
Hi 😄 so @kirsty14 makes a good point here, what you could do is create a `UITextField` `extension` something like this. 

```
extension UITextField {

   func applyStyle() {
      self.layer.borderColor = UIColor(named: "AppPrimaryColor")?.cgColor
   }
}
```

That way whenever you want to apply that styling to your `UITextField` you can just do this: 

`myTextField.applyStyle()`

Something you can also think about is adding that colour to a style kit 😄